src/Uno.UI/UI/Xaml/Controls/MenuFlyout/MenuFlyoutSubItem.cs

```diff
@@ -1,47 +1,888 @@
-	public  partial class MenuFlyoutSubItem : MenuFlyoutItemBase
+	public partial class MenuFlyoutSubItem : MenuFlyoutItemBase, ISubMenuOwner
 	{
-		public MenuFlyoutSubItem()
-		{
-			Items = new ObservableCollection<MenuFlyoutItemBase>();
-		}
+		// Popup for the MenuFlyoutSubItem
+		Popup m_tpPopup;
+
+		// Presenter for the MenuFlyoutSubItem
+		Control m_tpPresenter;
+
+		// In Threshold, MenuFlyout uses the MenuPopupThemeTransition.
+		Transition m_tpMenuPopupThemeTransition = null;
+
+		// Event pointer for the Loaded event
+		// IDisposable m_epLoadedHandler;
+
+		// Event pointer for the size changed on the MenuFlyoutSubItem's presenter
+		IDisposable m_epPresenterSizeChangedHandler;
+
+		// Helper to which to delegate cascading menu functionality.
+		CascadingMenuHelper m_menuHelper;
+
+		// Weak reference the parent that owns the menu that this item belongs to.
+		private WeakReference m_wrParentOwner;
 
-		public  string Text
+		DependencyObjectCollection<MenuFlyoutItemBase> m_tpItems;
+
+		public string Text
 		{
 			get => (string)this.GetValue(TextProperty);
 			set => SetValue(TextProperty, value);
 		}
 
-		public  global::System.Collections.Generic.IList<MenuFlyoutItemBase> Items
-		{
-			get;
-		}
+		public IList<MenuFlyoutItemBase> Items => m_tpItems;
 
-		public  IconElement Icon
+		public IconElement Icon
 		{
 			get => (IconElement)this.GetValue(IconProperty);
 			set => this.SetValue(IconProperty, value);
 		}
 
-		public static global::Windows.UI.Xaml.DependencyProperty TextProperty { get; } = 
+		public static Windows.UI.Xaml.DependencyProperty TextProperty { get; } =
 		Windows.UI.Xaml.DependencyProperty.Register(
 			"Text",
-			typeof(string), 
-			typeof(MenuFlyoutSubItem), 
+			typeof(string),
+			typeof(MenuFlyoutSubItem),
 			new FrameworkPropertyMetadata(default(string)));
 
-		public static global::Windows.UI.Xaml.DependencyProperty IconProperty { get; } = 
+		public static Windows.UI.Xaml.DependencyProperty IconProperty { get; } =
 		Windows.UI.Xaml.DependencyProperty.Register(
 			"Icon",
-			typeof(IconElement), 
-			typeof(MenuFlyoutSubItem), 
+			typeof(IconElement),
+			typeof(MenuFlyoutSubItem),
 			new FrameworkPropertyMetadata(default(IconElement)));
+
+		public MenuFlyoutSubItem()
+		{
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: ", this));
+#endif // MFSI_DEBUG
+
+			PrepareState();
+		}
+
+		void PrepareState()
+		{
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: PrepareState.", this));
+#endif // MFSI_DEBUG
+
+			// Create the sub menu items collection and set the owner
+			m_tpItems = new DependencyObjectCollection<MenuFlyoutItemBase>(this);
+
+			m_menuHelper = new CascadingMenuHelper();
+			m_menuHelper.Initialize(this);
+		}
+
+		void DisconnectFrameworkPeerCore()
+		{
+			// Ensure the clean up the items whenever MenuFlyoutSubItem is disconnected
+			//if (m_tpItems.GetAsCoreDO() != null)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5205492)